impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -153,363 +101,196 @@ public synchronized void stop() {
     }
   }
 
-  static void createLocationIfRequiredAndVerify(final File rootDirectory) {
-    if(!rootDirectory.exists()) {
-      if(!rootDirectory.mkdirs()) {
-        throw new IllegalArgumentException("Directory couldn't be created: " + rootDirectory.getAbsolutePath());
-      }
-    } else if(!rootDirectory.isDirectory()) {
-      throw new IllegalArgumentException("Location is not a directory: " + rootDirectory.getAbsolutePath());
-    }
-
-    if(!rootDirectory.canWrite()) {
-      throw new IllegalArgumentException("Location isn't writable: " + rootDirectory.getAbsolutePath());
-    }
+  File getLockFile() {
+    return lockFile;
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
-  public boolean handlesResourceType(ResourceType<?> resourceType) {
-    return ResourceType.Core.DISK.equals(resourceType);
-  }
+  public SafeSpaceIdentifier prepareSafeSpace(String owner, String identifier) {
```

Comment:
The name as _prepareSafeSpaceIdentifier_ or __createSafeSpaceIdentifier_, may be? Since it is just about creating the identifier? Just a personal opinion.